presto-main/src/main/java/com/facebook/presto/testing/LocalQueryRunner.java

```diff
@@ -838,6 +844,46 @@ private MaterializedResultWithPlan executeInternal(Session session, @Language("S
         }
     }
 
+    private MaterializedResultWithPlan executeExplainTypeValidate(String sql, Session session, WarningCollector warningCollector)
+    {
+        AnalyzerOptions analyzerOptions = createAnalyzerOptions(session, warningCollector);
+        BuiltInPreparedQuery preparedQuery = new BuiltInQueryPreparer(sqlParser).prepareQuery(analyzerOptions, sql, session.getPreparedStatements(), warningCollector);
+        assertFormattedSql(sqlParser, createParsingOptions(session), preparedQuery.getStatement());
+
+        PlanNodeIdAllocator idAllocator = new PlanNodeIdAllocator();
+
+        // TODO: Is this needed?
```

Comment:
Can you figure out the answer and remove the todo?

Reply:
At this point this code base is borrowed from existing methods. I was trying to point out that Type Validate should not need Explainer class. I will remove the TODO for now, as it is improvement for later.